hotfix/B3000-6

nikikorn opened the pull request:
Variables now NOT static.
Now using directory of .jar storage.
try-catch block was changed.

nikikorn committed:
Variables now NOT static.
Now using directory of .jar storage.
try-catch block was changed.

Athen1an commented on src/main/java/ru/nikikorn/backup/BackupServiceApplication.java:
```diff
@@ -3,15 +3,26 @@
+
+
+
+
```
Это чего за куча пустых строк?

nikikorn replied:
это я тестил что-то, все пустые строки в конце уберу)

Athen1an commented on src/main/java/ru/nikikorn/backup/BackupSettings.java:
```diff
@@ -1,70 +1,51 @@
+    private boolean includeDirectory;
+    private Integer backupCicleSize;
+
+    public void start() {
```
Сделай не метод start, а чтобы инициализация происходила при создании объекта

Athen1an commented on src/main/java/ru/nikikorn/backup/BackupSettings.java:
```diff
@@ -1,70 +1,51 @@
             targetDirectory = "/Users/nikikorn/IdeaProjects/Backup-service/";
             backupDirectory = "/Users/nikikorn/IdeaProjects/Backup-service/backup/";
             includeDirectory = false;
             backupCicleSize = 3;
```
У тебя код повтаряется, вынеси инициализацию переменных за блоки трай и кетч. По факту, у тебя в трай просто пытается считаться файл - не нашел, просто пишем лог. А дальше пытаемся засетить переменные из объекта properties, если есть - ок, нет - берем дефолтовые.

Athen1an commented on src/main/java/ru/nikikorn/backup/BackupSettings.java:
```diff
@@ -1,70 +1,51 @@
-                FileNotFoundException e) {
-            System.out.println("Configuration file was not found - the default settings was applied.");
+            targetDirectory = properties.getProperty("targetDirectory","/Users/nikikorn/IdeaProjects/Backup-service/target/");
+            backupDirectory = properties.getProperty("backupDirectory","/Users/nikikorn/IdeaProjects/Backup-service/target/backup/");
```
Брать дефолтовые значений из пути, где лежит jar архив. Сейчас они у тебя хардкодом вбиты

nikikorn committed:
Variables initialization now in constructor.
Now using directory of .jar storage.(not hard code)
try-catch block was changed. Variables outside try-catch

.jar successfully created and can be open.

Athen1an commented on src/main/java/ru/nikikorn/backup/BackupSettings.java:
```diff
@@ -1,70 +1,48 @@
+    private String backupDirectory;
+    private boolean includeDirectory;
+    private Integer backupCicleSize;
+    private String  jar = Path.of(BackupServiceApplication.class.getProtectionDomain().getCodeSource().getLocation().getPath()).getParent().normalize() +"/";
```
Перенеси эту переменную в конструктор. Ты ее больше нигде не используешь, она не нужна объекту этого класса.

Athen1an replied:
И сделай нормализацию пробелов\строк в проекте (ctr clt l), у тебя после String два пробела

nikikorn replied:
Готово!

Athen1an commented:
И ты так и не удалил target директорию из проекта

nikikorn committed:
Variables initialization now in constructor.
Now using directory of .jar storage.(not hard code)
try-catch block was changed. Variables outside try-catch

.jar successfully created and can be open.
